core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/registry/AbstractTicketRegistry.java

```diff
@@ -99,7 +106,16 @@ public Ticket getTicket(final String ticketId) {
                 val ticketAgeSeconds = getTicketAgeSeconds(ticket);
                 LOGGER.debug("Ticket [{}] has expired according to policy [{}] after [{}] seconds and [{}] uses and will be removed from the ticket registry",
                     ticketId, ticket.getExpirationPolicy().getName(), ticketAgeSeconds, ticket.getCountOfUses());
-                deleteSingleTicket(ticket);
+                if (ticket instanceof final TicketGrantingTicket tgt) {
+                    val clientInfo = ClientInfoHolder.getClientInfo();
+                    applicationContext.publishEvent(new CasRequestSingleLogoutEvent(this, tgt, clientInfo));
+                    applicationContext.publishEvent(new CasTicketGrantingTicketDestroyedEvent(this, tgt, clientInfo));
+                }
+                try {
+                    deleteTicket(ticket);
+                } catch (final Exception e) {
+                    LOGGER.warn("Deletion failed for expired ticket [{}]", ticket.getId());
```

Comment:
This should use `LoggingUtils` to log the warn. Please do not "eat"" the exception.

Reply:
Good point. Will change.